cmd/kueuectl/app/util/dry_run.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog/v2"
+)
+
+type DryRunStrategy int
+
+const (
+	// DryRunNone indicates the client will make all mutating calls
+	DryRunNone DryRunStrategy = iota
+
+	// DryRunClient, or client-side dry-run, indicates the client will prevent
+	// making mutating calls such as CREATE, PATCH, and DELETE
+	DryRunClient
+
+	// DryRunServer, or server-side dry-run, indicates the client will send
+	// mutating calls to the APIServer with the dry-run parameter to prevent
+	// persisting changes.
+	//
+	// Note that clients sending server-side dry-run calls should verify that
+	// the APIServer and the resource supports server-side dry-run, and otherwise
+	// clients should fail early.
+	//
+	// If a client sends a server-side dry-run call to an APIServer that doesn't
+	// support server-side dry-run, then the APIServer will persist changes inadvertently.
+	DryRunServer
+)
+
+// AddDryRunFlag adds dry-run flag to a command.
+func AddDryRunFlag(cmd *cobra.Command) {
+	cmd.Flags().String(
+		"dry-run",
+		"none",
+		`Must be "none", "server", or "client". If client strategy, only print the object that would be sent, without sending it. If server strategy, submit server-side request without persisting the resource.`,
+	)
+}
+
+func GetFlagString(cmd *cobra.Command, flag string) string {
```

Comment:
```suggestion
func getFlagString(cmd *cobra.Command, flag string) string {
```

Reply:
Maybe better to move this function to another file? I think it will useful in future.

Reply:
possibly, but we can leave it for a follow up, when we see more use cases.

Reply:
I propose to keep it public that we can use this function in another packages if it will need.